### PR TITLE
feat: add support to display examples

### PR DESCRIPTION
**Pull request by ITman1**
**Description**

- When `example` attribute is specified alongside with type/record field definition it is propagated into `examples` attribute of AsyncAPI schema 
- Tested locally and on REACT component playground

image

**Related issue(s)**
Implements `example` feature from https://github.com/asyncapi/avro-schema-parser/issues/18

### CODE REVIEW

**Comment by derberg** [comment]
@ITman1 please update with latest master

**Comment by ITman1** [comment]
@derberg Updated from upstream repo

**Comment by bpb54321** [comment]
When will this be merged? I would like to use this ASAP in my project.

**Comment by derberg** [comment]
@bpb54321 I want to wait at least until the end of the week. Maybe @iemejia called out by @ITman1 will have time to have a look and will share his opinion on this. Otherwise next week definitely.

**Comment by derberg** [comment]
@ITman1 please have a look at failing linter, it complains on the complexity of the code

**Comment by ITman1** [comment]
@derberg Tried to extract some code into seperate functions, Sonar does not complain anymore locally. Both tests started to fail after my change so I also adjusted expected test results. IMHO there was a bug, Avro can have 'null' as default values, before null was ignored from the spec files and so does also 0 I would guess.

File: to-json-schema.js [hunk 1]
```diff
@@ -21,17 +21,52 @@ const typeMappings = {
...
   const jsonSchema = {};
   const isUnion = Array.isArray(avroDefinition);
 
   if (isUnion) {
     jsonSchema.oneOf = [];
+    let nullDef = null;
     for (const avroDef of avroDefinition) {
       const def = await avroToJsonSchema(avroDef);
-      jsonSchema.oneOf.push(def);
+      const defType = avroDef.type || avroDef;
```
Comment (derberg) [inline to-json-schema.js:63]:
```suggestion
      // avroDef can be { type: 'int', default: 1 } and this is why avroDef.type has priority here
      const defType = avroDef.type || avroDef;
```

sorry but I need such comments in code as it is hard to maintain such project overtime, where changes are done very rarely

**Review by derberg — CHANGES_REQUESTED** [review]
one tiny request to add one additional comment to the code + please update readme with information of extra fields support

**Comment by M3lkior** [comment]
> @ITman1 fyi, I asked the community to share their opinion on a related issue as I'm not sure if we should actually provide such a support

From my side, it could be nice to have this feature merged. Here is the result of my generation based on `avsc` files ; all the attributes are null ; handle the `example` attribute could improve the reader experience ;)

image

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@ITman1 hey, sorry I did not notice you made required changes, if @M3lkior would not add his comment, I would miss it. There are too many PRs in many different repos. Next time you contribute (I hope you will 😄 ) please use the button to `re-request` review to trigger a notification.

**Pull request merged**